src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -945,4 +939,59 @@ private void writeHeaderAndBadVariant(final VariantContextWriter writer) {
         writer.add(vc);
     }
 
+    final String baseVariants = packageRootTestDir + "engine/feature_data_source_test.vcf";
+
+    @CommandLineProgramProperties(programGroup = TestProgramGroup.class, oneLineSummary = "GATKTool Intervals Test Walker", summary = "This is a test walker for GATKTool getIntervals")
+    private static class TestIntervalWalker extends GATKTool {
+        @Argument(fullName = StandardArgumentDefinitions.VARIANT_LONG_NAME, shortName = StandardArgumentDefinitions.VARIANT_SHORT_NAME, doc = "A VCF file containing variants", common = false, optional = false)
+        public String drivingVariantFile;
+
+        public TestIntervalWalker() {
+
+        }
+
+        @Override
+        public void traverse() {
+
+        }
+    }
+
+    @DataProvider(name = "TestGetIntervalsProvider")
+    public Object[][] getTestGetIntervalsProvider() {
+        return new Object[][]{
+                {"1:21-21", 1, hg19MiniReference, false},
+                {null, 4, hg19MiniReference, false},
+                {"1:21-21", 1, null, true},  //this will fail with an error
```

Comment:
This is an excellent test case, but its out of place in this provider since the test fails well before it gets to `getTraversalIntervals`. It should come out of this provider but it would be worth keeping as a separate `testSequenceDictionaryRequiredForIntervalQuery` test method.